fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -706,6 +708,29 @@ public Void visitFunctionCall(FunctionCallExpr node, Scope scope) {
                     throw new SemanticException(
                             FunctionSet.TIME_SLICE + " requires second parameter must be greater than 0");
                 }
+                if (node.getChild(0) instanceof StringLiteral) {
```

Comment:
what if the first arg is a string column ?  we should support the timeslice("varchar"), then handle this logic in BE.

Reply:
> what if the first arg is a string column ? we should support the timeslice("varchar"), then handle this logic in BE.
string column is supported by implicit cast from varchar to date/datetime.
This codes just for input of text(not date/datetime column) like "2014-12-21 12:34:56", because we can only look into text to find right type(datetime or date).

Reply:
> what if the first arg is a string column ? we should support the timeslice("varchar"), then handle this logic in BE.

string column is supported by implicit cast from varchar to date/datetime.
This codes just for input of text(not date/datetime column) like "2014-12-21 12:34:56", because we can only look into text to find right type(datetime or date).